xdsl/backend/csl/print_csl.py

```diff
@@ -0,0 +1,170 @@
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import IO
+
+from xdsl.dialects import arith, func
+from xdsl.dialects.builtin import (
+    Float16Type,
+    Float32Type,
+    FloatAttr,
+    IndexType,
+    IntAttr,
+    IntegerAttr,
+    IntegerType,
+    ModuleOp,
+    Signedness,
+    SignednessAttr,
+)
+from xdsl.ir import Attribute, Block, SSAValue
+
+
+@dataclass
+class CslPrintContext:
+    output: IO[str]
+
+    variables: dict[SSAValue, str] = field(default_factory=dict)
+
+    _counter: int = field(default=0)
```

Comment:
my first instinct is that it would be neater to have a counter per name, to avoid
```
a0
b1
c2
```
```suggestion
    _counter: int = field(default_factory=lambda: defaultdict(counter))
```

Reply:
Ah, I see the problem, the potential clashing is annoying. I'm not sure what the right way to deal with that is, everything that doesn't involve a global analysis of all possible names seems broken, so might as well go with your simple solution

Reply:
Yeah, almost every aspect of this needs improving. This is a "good enough" thing to get the ball rolling